learn/work/sparse-recipe/index.qmd

```diff
@@ -0,0 +1,254 @@
+
+We are furthermore assuming that this prediction will take place at departure time. This means we have to exclude some variables as they contain information that is not yet available.
+
+```{r}
+flights <- flights |>
+  select(-arr_time, -air_time)
+```
+
+This data set contains a number of redundant variables. We don't need to know the departure time `dep_time`, scheduled departure time `sched_dep_time`, and the departure delay `dep_delay` as they are a linear combination of each other `dep_delay = dep_time - sched_dep_time`. So we can remove one of them, we will choose `sched_dep_time`.
+
+Likewise, the `time_hour` variable is a datetime that contains data also located in `year`, `month`, `day`, `hour`, and `minute`. We will thus also remove that one.
+
+```{r}
+flights <- flights |>
+  select(-time_hour, -sched_dep_time)
+```
+
+You may or may not have noticed that `dep_time` and `sched_arr_time` have a weird encoding. What is happening is that `517` is actually `5:17` e.i. 17 minutes past 5 AM. So we need to update that, which we will use a little helper function for.
+
+```{r}
+get_minutes <- function(x) {
+ minutes <- x %% 100
+ hours <- x %/% 100
+
+ hours * 60 + minutes
+}
+
+flights <- flights |>
+  mutate(across(c(dep_time, sched_arr_time), get_minutes))
+```
+
+We will fit a model using the first month of the year, and then try to assess how well it will generalize over the remaining years. We will also exclude any observations where `arr_delay` is `NA`.
+
+```{r}
+flights <- drop_na(flights, arr_delay)
+flights_train <- filter(flights, month == 1)
+flights_test <- filter(flights, month != 1)
+```
+
+## Creating a recipe
+
+The data is quite simple in terms of types. We have numeric variables and categorical variables. We will do some simple imputation of the numeric variables and create dummy variables on the categorical predictors.
+
+```{r}
+rec_spec <- recipe(arr_delay ~ ., data = flights_train) |>
+  step_impute_mean(all_numeric_predictors()) |>
+  step_zv(all_predictors()) |>
+  step_normalize(all_numeric_predictors()) |>
+  step_novel(all_nominal_predictors()) |>
+  step_unknown(all_nominal_predictors()) |>
+  step_dummy(all_nominal_predictors())
+
+rec_spec
+```
+
+You will notice that we aren't doing anything special here to denote the recipe to produce sparsity. Next will go into some details to explain how you the user should approach recipes when you suspect that sparsity will be produced.
+
+## How is sparsity handled in recipes
+
+There have been made two types of changes to recipes steps with regards to sparsity. First, a number of steps have gained a `sparse` argument. This is used to toggle the creation of sparse vectors. `step_dummy()` that we used in our recipe above is one of such steps.
```

Comment:
```suggestion
There have been made two types of changes to recipes steps regarding sparsity. 

First, some steps can augment the data set with many columns that are naturally sparse. Creating binary indicators from a factor predictor, via `step_dummy()`, is a good example. Because of this, a number of steps have gained a `sparse` argument, which toggles the creation of sparse vectors. 
```